File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s867221485_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	var N int
	_, _ = fmt.Sscan(scanner.Text(), &N)
	scanner.Scan()
	s := scanner.Text()

	ans := 0
	sha := 0
	dot := 0
	first := true

	for i := 0; i < len(s); i++ {
		c := s[i]
		if c == '#' {
			first = false
			if sha <= dot {
				ans += sha
				sha = 0
				dot = 0
				first = true
			}
			sha++
		} else if c == '.' {
			if !first {
				dot++
			}
		}
	}

	if sha <= dot {
		ans += sha
		dot = 0
	}

	fmt.Println(ans + dot)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s867221485_commented.py
import sys

def main():
    N = int(sys.stdin.readline())
    s = sys.stdin.readline().strip()
    ans = 0
    sha = 0
    dot = 0
    first = True
    for c in s:
        if c == '#':
            first = False
            if sha <= dot:
                ans += sha
                sha = 0
                dot = 0
                first = True
            sha += 1
        else:
            if not first:
                dot += 1
    if sha <= dot:
        ans += sha
        dot = 0
    print(ans + dot)

if __name__ == "__main__":
    main()
# End of Code
